arctic/store/version_store.py

```diff
@@ -107,7 +115,7 @@ def _read_preference(self, allow_secondary):
         return ReadPreference.NEAREST if allow_secondary else ReadPreference.PRIMARY
 
     @mongo_retry
-    def list_symbols(self, all_symbols=False, snapshot=None, regex=None, **kwargs):
+    def list_symbols(self, all_symbols=False, snapshot=None, regex=None, metadata=False, **kwargs):
```

Comment:
because the option to list symbols only with point in time metadata is going to be mostly used by/for dev:
1) consider renaming the parameter to timed_metadata_only, name=metadata is ambiguous
2) consider hiding it in kwargs ? ( = advanced dev usage only )

As Yoav said this should probably behave like a meta(...) search